src/lib/Syntax.hs

```diff
@@ -215,6 +215,7 @@ data UExpr' = UVar UVar
             | UFor Direction UPatAnn UExpr
             | UCase UExpr [UAlt]
             | UHole
+            | UAnnotatedExpr UExpr UExpr
```

Comment:
nit: `UTypeAnn` seems to be a nice short name, but I don't mind keeping this one.